src/main/scala/dsptools/numbers/chisel_types/DspComplexTypeClass.scala

```diff
@@ -12,30 +12,51 @@ class DspComplexRing[T <: Data:Ring] extends Ring[DspComplex[T]] with hasContext
   def plus(f: DspComplex[T], g: DspComplex[T]): DspComplex[T] = {
     DspComplex.wire(f.real + g.real, f.imag + g.imag)
   }
+  def plusContext(f: DspComplex[T], g: DspComplex[T]): DspComplex[T] = {
+    DspComplex.wire(f.real context_+ g.real, f.imag context_+ g.imag)
+  }
   def times(f: DspComplex[T], g: DspComplex[T]): DspComplex[T] = {
+    val c_p_d = g.real + g.imag
```

Comment:
I'm feeling weak on my complex arithmetic, @grebe can you template this out for me a little if it's important